### ISSUE TITLE
Fix errors from pylint/mypy

### ISSUE THREAD

**Issue by LewisGaul** [LewisGaul/minegauler#158]

**Comment by deepgandhi** [comment]
Need more details and discription about location of file and error.

**Comment by LewisGaul** [comment]
Hey @deepgandhi, I've just pushed a pylint config file (use the `dev` branch as the base).

You can find errors by running e.g.:
```
pip install pylint
pylint minegauler/shared/utils.py
```
This shows only the errors in the specified file, in this case:
```
$pylint minegauler/shared/utils.py
************* Module minegauler.shared.utils
minegauler/shared/utils.py:115:2: W0511: @@@LG Some attention please :) (fixme)
minegauler/shared/utils.py:88:12: W0612: Unused variable 'j' (unused-variable)
minegauler/shared/utils.py:181:12: C0200: Consider using enumerate instead of iterating with range and len (consider-using-enumerate)
minegauler/shared/utils.py:57:0: W0611: Unused import paths (unused-import)
```
Any of those errors are open for fixing.

In general the 'fixme' warnings might not be easy to sort, and there will be some false positives. If you're unsure about any feel free to ask here or in GitHub Discussions.

**Comment by deepgandhi** [comment]
Does pylint is to be checked with all files?

**Comment by LewisGaul** [comment]
> Does pylint is to be checked with all files?

Yep pylint can be used to find suggestions for cleaner code in any of the files. I'd probably suggest focusing on `minegauler/core/` first. I've left a review on your PR that would be good to address first.

**Comment by LewisGaul** [comment]
Updating this to now prefer ruff rather than pylint (added ruff config in `pyproject.toml`)
